keanu-project/src/main/java/io/improbable/keanu/vertices/VertexLabel.java

```diff
@@ -37,18 +46,18 @@ public boolean isInNamespace(String... namespace) {
     }
 
     public VertexLabel withExtraNamespace(String topLevelNamespace) {
-        List<String> newNamespace = ImmutableList.<String>builder().addAll(namespace).add(topLevelNamespace).build();
-        return new VertexLabel(this.name, newNamespace);
+        List<String> newNamespace = ImmutableList.<String>builder().add(topLevelNamespace).addAll(namespace).build();
+        return new VertexLabel(newNamespace, this.name);
     }
 
     public VertexLabel withoutOuterNamespace() {
-        List<String> reducedNamespace = namespace.subList(0, namespace.size() - 1);
-        return new VertexLabel(this.name, reducedNamespace);
+        List<String> reducedNamespace = namespace.subList(1, namespace.size());
+        return new VertexLabel(reducedNamespace, this.name);
     }
 
     public Optional<String> getOuterNamespace() {
         try {
-            return Optional.of(namespace.get(namespace.size() - 1));
+            return Optional.of(namespace.get(0));
```

Comment:
You can use `return namespace.stream().findFirst();` instead of the try catch block.